### DISCUSSION TITLE
[Bug] difference of kv-cache-prefixing between vLLM and sglang

### DISCUSSION THREAD

**Discussion by chenchunhui97** [sgl-project/sglang#1669] [General]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

no bug. I am just wondering the difference of kv-cache-prefixing between vLLM impletention and SGLang implementation.

vLLM use hash to store and verify cached token:
<img width="318" alt="image" src="https://github.com/user-attachments/assets/c89b6dc1-101a-43de-b925-8ba1e45e6d20">
SGLang uses RadixAttention, so what is the difference? I found SGLang is faster than vLLM, why SGLang RadixAttention is faster than vLLM KV-Cache-prefixing?

### Reproduction

not available

### Environment

//

**Comment by zhyncs** [comment]
https://github.com/vllm-project/vllm/issues/2614
